modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/jaxrs/exception/mapper/WorkflowExceptionMapper.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.headless.commerce.delivery.cart.internal.jaxrs.exception.mapper;
+
+import com.liferay.commerce.exception.CommerceWorkflowException;
+import com.liferay.headless.commerce.core.exception.mapper.BaseExceptionMapper;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Mahmoud Azzam
+ */
+@Component(
+	property = {
+		"osgi.jaxrs.application.select=(osgi.jaxrs.name=Liferay.Headless.Commerce.Delivery.Cart)",
+		"osgi.jaxrs.extension=true",
+		"osgi.jaxrs.name=Liferay.Headless.Commerce.Delivery.Cart.WorkflowExceptionMapper"
+	},
+	service = ExceptionMapper.class
+)
+@Provider
+public class WorkflowExceptionMapper
+	extends BaseExceptionMapper<CommerceWorkflowException> {
+
+	@Override
+	public String getErrorDescription() {
+		return "Order is already assigned";
+	}
+
+	@Override
+	public Response.Status getStatus() {
+		return Response.Status.BAD_REQUEST;
```

Comment:
not sure this is the pattern we agreed with Brian, please reach out to  @alessiorendina and ask him.

Reply:
anyway I think you should add this also in the admin API

Reply:
I guess the admin should be able to change the order regardless of any active workflow

Reply:
The new pattern is described here: https://issues.liferay.com/browse/LPS-181443